### PR TITLE
docs(samples): new Doc AI samples for v1beta3

### PR DESCRIPTION
**Pull request by aribray**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-documentai/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: samples/batch_process_documents_sample_v1beta3.py
Comment (telpirion) [inline samples/batch_process_documents_sample_v1beta3.py:104]: Need `[END documentai_batch_process_document]` after this line

File: samples/process_document_sample_v1beta3.py
Comment (telpirion) [inline samples/process_document_sample_v1beta3.py:24]: Drop 'sample' in the region tag, e.g.:

`[START documentai_process_document]`

File: samples/quickstart_sample_v1beta3.py
Comment (telpirion) [inline samples/quickstart_sample_v1beta3.py:23]: Drop 'sample' here too, e.g. `[START documentai_quickstart]`

File: samples/process_document_sample_v1beta3.py
Comment (telpirion) [inline samples/process_document_sample_v1beta3.py:20]: Remove this from the public sample--that processor only exists in our GCP project

File: samples/batch_process_documents_sample_v1beta3.py
Comment (telpirion) [inline samples/batch_process_documents_sample_v1beta3.py:22]: Remove this from public sample.

File: samples/batch_process_documents_sample_v1beta3.py
Comment (telpirion) [inline samples/batch_process_documents_sample_v1beta3.py:24]: Remove this from public sample.

File: samples/batch_process_documents_sample_v1beta3.py
Comment (telpirion) [inline samples/batch_process_documents_sample_v1beta3.py:26]: Remove this from public sample--customers may not have access to this bucket.

File: samples/quickstart_sample_v1beta3.py
Comment (telpirion) [inline samples/quickstart_sample_v1beta3.py:20]: Remove from public sample

File: samples/quickstart_sample_v1beta3.py
Comment (telpirion) [inline samples/quickstart_sample_v1beta3.py:21]: Remove

File: tests/unit/gapic/documentai_v1beta3/quickstart_v1beta3_test.py
Comment (telpirion) [inline tests/unit/gapic/documentai_v1beta3/quickstart_v1beta3_test.py:37]: I think we might want to get the `project_id` from environment variables.
```
project_id = os.environs['GCLOUD_PROJECT']
```

There may be a niftier way to get this from the Document AI client (but I don't know it off the top of my head).

File: tests/unit/gapic/documentai_v1beta3/quickstart_v1beta3_test.py
Comment (telpirion) [inline tests/unit/gapic/documentai_v1beta3/quickstart_v1beta3_test.py:46]: Compose this at runtime using the `project_id` and `processor_id`.

File: tests/unit/gapic/documentai_v1beta3/quickstart_v1beta3_test.py
Comment (telpirion) [inline tests/unit/gapic/documentai_v1beta3/quickstart_v1beta3_test.py:39]: You shouldn't need any GCS uri stuff for the quickstart.

File: tests/unit/gapic/documentai_v1beta3/quickstart_v1beta3_test.py
Comment (telpirion) [inline tests/unit/gapic/documentai_v1beta3/quickstart_v1beta3_test.py:24]: Remove (unneeded) service imports.

**Review by telpirion — COMMENTED** [review]
I think that the sample test are in the wrong place. We typically place the tests in the same folder as the script under test.

E.g. `quickstart_test.py` is in the same folder as `quickstart.py`

File: samples/batch_process_documents_sample_v1beta3.py
Comment (telpirion) [inline samples/batch_process_documents_sample_v1beta3.py:38]: We're not parsing forms (solely) in this sample--please remove.

File: samples/process_document_sample_v1beta3.py
Comment (telpirion) [inline samples/process_document_sample_v1beta3.py:28]: This line may not be needed. If not, let's drop it. Instead you can just instantiate a client by doing:

```
client = documentai.DocumentProcessorServiceClient()
```

File: samples/quickstart_sample_v1beta3.py
Comment (telpirion) [inline samples/quickstart_sample_v1beta3.py:28]: Same as above--instantiate the client without options if you can.

File: samples/process_document_sample_v1beta3.py
Comment (aribray) [reply samples/process_document_sample_v1beta3.py]: You were right, it was unneeded. I removed it

File: samples/quickstart_sample_v1beta3.py
Comment (telpirion) [inline samples/quickstart_sample_v1beta3.py:55]: Always provide full brand name for products.

Comment (telpirion) [reply samples/quickstart_sample_v1beta3.py:55]: We're also not dealing with just form fields any more.

**Comment by leahecole** [comment]
putting a do not merge on it right away because the samples checks aren't running - I'll look into that because I'm like 99% sure it's not your fault Ari :)

**Comment by leahecole** [comment]
blocked by internal CL 337396830 to make checks run

**Comment by leahecole** [comment]
Alright - CL merged, let's see if the checks show up

**Comment by busunkim96** [comment]
Hi @aribray,

Our documentation on this is not great at the moment, but the tests expect directories nested under samples. If these are generic snippets please move them into `samples/snippets`.

Example: https://github.com/googleapis/python-automl/tree/master/samples/snippets

You'll also need to copy in a `noxfile.py` next to the samples. You can copy the contents from https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/noxfile-template.py

**Comment by leahecole** [comment]
Thank you Bu Sun!

File: samples/tests/quickstart_v1beta3_test.py
Comment (leahecole) [inline samples/tests/quickstart_v1beta3_test.py:53]: to use the fixture, you then would call it in the function, so your function declaration would look like `def test_quickstart(capsys, test_bucket)`

File: samples/tests/quickstart_v1beta3_test.py
Comment (leahecole) [inline samples/tests/quickstart_v1beta3_test.py:54]: Question - I'm not 100% sure where you use the bucket that you're creating - is it part of the filepath?

File: samples/tests/quickstart_v1beta3_test.py
Comment (leahecole) [inline samples/tests/quickstart_v1beta3_test.py:56]: Is there a particular reason you're using `sys.stdout` rather than just printing? I know it's more fine grained logging, which I do prefer in general tbh, but I think we don't consistently use that in other samples.

File: samples/tests/quickstart_v1beta3_test.py
Comment (leahecole) [inline samples/tests/quickstart_v1beta3_test.py:32]: Very interested in this `os.getcwd()` here - can we not assume that the user will be running the tests from this directory? I'm thinking it might be overkill and that we can make some assumptions, but I also could be totally off base and absolutely welcome further discussion here

File: samples/batch_process_documents_sample_v1beta3.py
Comment (leahecole) [inline samples/batch_process_documents_sample_v1beta3.py:76]: What is el? Definitely prefer using a more descriptive varaible name

Comment (leahecole) [reply samples/batch_process_documents_sample_v1beta3.py:76]: Also, is there a strong reason why this needs to be protected? Tbh, I'm not super picky about protected/private methods in Python (maybe I should be) and am mainly curious to hear the thought process around it

**Review by leahecole — CHANGES_REQUESTED** [review]
Meta comments 

1) in python, we typically put the tests alongside the the samples - so when you're doing the reorg Bu Sun asked for - putting the samples in a `snippets` folder, you can move the tests into that same folder. Each test should have the same name as its corresponding snippet, but with the `_test` suffix. For example, if your sample was `do_awesome_doc_thing.py` your test would be `do_awesome_doc_thing_test.py`
2) For both samples and tests, comments I made on one file typically apply to all related ones (for example, the fixtures - my comments would apply to all of the setup/teardown spots) - I decided commenting the same thing on 3 different files was overkill, but if it would help you for me to go back and mark em up, let me know and I'm happy to copy paste :) 
3) Python fixtures are kinda tough at first - they behave totally differently than node and I know I always have to shut my JS brain off when writing Python tests. These [translate tests](https://github.com/googleapis/python-translate/blob/master/samples/snippets/beta_snippets_test.py) do a good job of showing how they spin up and tear down a bucket (as well as other resources) that they use in testing

File: samples/tests/quickstart_v1beta3_test.py
Comment (aribray) [reply samples/tests/quickstart_v1beta3_test.py:35]: Thanks! I'll try it this way

File: samples/tests/quickstart_v1beta3_test.py
Comment (aribray) [reply samples/tests/quickstart_v1beta3_test.py:54]: This is a good question - and I'm realizing now that I carried these fixtures over from the `batch_process` tests. These tests don't have anything to tear down :)

File: samples/tests/quickstart_v1beta3_test.py
Comment (aribray) [reply samples/tests/quickstart_v1beta3_test.py:36]: You're right, I don't think it's necessary. I'll remove!

File: samples/batch_process_documents_sample_v1beta3.py
Comment (telpirion) [reply samples/batch_process_documents_sample_v1beta3.py]: "el" for "element" (probably should be renamed).

You bring up a good point--this function isn't protected for any strong reason besides keeping it local to where it's used. That said, it's used in each of the samples. Maybe it's a good time to break this out into a utility function?

Comment (aribray) [reply samples/batch_process_documents_sample_v1beta3.py]: No strong reason - I'll edit

File: samples/tests/quickstart_v1beta3_test.py
Comment (aribray) [reply samples/tests/quickstart_v1beta3_test.py:29]: I think it makes sense to make that assumption, especially based off of similar snippets. I'll adjust.

File: samples/snippets/batch_process_documents_sample_v1beta3.py
Comment (leahecole) [inline samples/snippets/batch_process_documents_sample_v1beta3.py:76]: `el` needs to be a more descriptive variable name

File: samples/snippets/batch_process_documents_sample_v1beta3.py
Comment (leahecole) [inline samples/snippets/batch_process_documents_sample_v1beta3.py:96]: out of curiosity, are these all of the fields someone can print? If not, might be worth linking in a comment to the docs for the other fields that someone could print

File: samples/snippets/batch_process_documents_sample_v1beta3_test.py
Comment (leahecole) [inline samples/snippets/batch_process_documents_sample_v1beta3_test.py:42]: it's not good practice to catch `Exception` here because it can mask legitimate errors that get through. Instead, use a specific `Exception` here (likely some variation of `NotFound`)

File: samples/snippets/batch_process_documents_sample_v1beta3_test.py
Comment (leahecole) [inline samples/snippets/batch_process_documents_sample_v1beta3_test.py:43]: not good practice to use `pass` as anything other than a placeholder - instead, add a print statement that tells the user why this exception is OK. Here it'll be something like `print("Bucket already deleted")`

File: samples/snippets/process_document_sample_v1beta3.py
Comment (leahecole) [inline samples/snippets/process_document_sample_v1beta3.py:52]: same note about `el` needing to be more descriptive

File: samples/snippets/process_document_sample_v1beta3.py
Comment (leahecole) [inline samples/snippets/process_document_sample_v1beta3.py:70]: question why are we only looking at the first page here? Or am I misunderstanding what's going on?

File: samples/snippets/batch_process_documents_sample_v1beta3.py
Comment (leahecole) [reply samples/snippets/batch_process_documents_sample_v1beta3.py:76]: Also, I'm noticing that this is a nested function, and I am skeptical about using nested functions - let's talk about it when we meet on Monday :)

File: samples/snippets/process_document_sample_v1beta3.py
Comment (leahecole) [reply samples/snippets/process_document_sample_v1beta3.py:52]: and nested functions

File: samples/snippets/process_document_v1beta3_test.py
Comment (leahecole) [inline samples/snippets/process_document_v1beta3_test.py:21]: I think some of the variables here aren't used

File: samples/snippets/quickstart_sample_v1beta3.py
Comment (leahecole) [inline samples/snippets/quickstart_sample_v1beta3.py:54]: same note about `el` and nested functions

File: samples/snippets/quickstart_sample_v1beta3.py
Comment (leahecole) [inline samples/snippets/quickstart_sample_v1beta3.py:68]: same question about first page

File: samples/snippets/quickstart_v1beta3_test.py
Comment (leahecole) [inline samples/snippets/quickstart_v1beta3_test.py:21]: same note about double checking unused variables

**Review by leahecole — CHANGES_REQUESTED** [review]

**Review by leahecole — CHANGES_REQUESTED** [review]
Two other meta notes - I forgot to ask you to add a `requirements.txt` and `requirements-test.txt` files with appropriate dependencies. The `requirements.txt` should have all packages a user needs to install before they can run your samples (pinned explicitly to specific versions), and the `requirements-test.txt` needs to have all of the packages a user would need to run a test - it may just be a one-liner that says `pytest==6.1.1`.

File: samples/snippets/batch_process_documents_sample_v1beta3.py
Comment (aribray) [reply samples/snippets/batch_process_documents_sample_v1beta3.py:76]: I'm using the `[v1beta2` samples](https://cloud.google.com/document-ai/docs/form-parser#v1beta2) as a reference for the nested function

File: samples/snippets/batch_process_documents_sample_v1beta3.py
Comment (aribray) [reply samples/snippets/batch_process_documents_sample_v1beta3.py:96]: Good idea to link to the documentation.

File: samples/snippets/process_document_sample_v1beta3.py
Comment (aribray) [reply samples/snippets/process_document_sample_v1beta3.py]: Resolved!

**Review by telpirion — APPROVED** [review]
LGTM pending other reviewer feedback.

File: samples/snippets/batch_process_documents_sample_v1beta3.py
Comment (leahecole) [inline samples/snippets/batch_process_documents_sample_v1beta3.py:101]: 😍  omg type hints

File: samples/snippets/batch_process_documents_sample_v1beta3_test.py
Comment (leahecole) [inline samples/snippets/batch_process_documents_sample_v1beta3_test.py:31]: Is there a reason why the output is going to a static bucket instead of just creating a bucket that you delete at the end?

File: samples/snippets/process_document_sample_v1beta3_test.py
Comment (leahecole) [inline samples/snippets/process_document_sample_v1beta3_test.py:24]: Is this filepath for suresies right?

**Review by leahecole — CHANGES_REQUESTED** [review]
A couple of hopefully quick things! We're so close!

File: samples/snippets/batch_process_documents_sample_v1beta3_test.py
Comment (aribray) [reply samples/snippets/batch_process_documents_sample_v1beta3_test.py:31]: I used the same process as the Node samples, but I'll ask Eric about it. Now that you say it, it does make sense to just tear it down at the end.

File: samples/snippets/process_document_sample_v1beta3_test.py
Comment (aribray) [reply samples/snippets/process_document_sample_v1beta3_test.py:24]: It was for SURE overkill. Resolved!

File: samples/snippets/batch_process_documents_sample_v1beta3_test.py
Comment (aribray) [reply samples/snippets/batch_process_documents_sample_v1beta3_test.py:31]: I'm back! I was deleting the files in the bucket, but forgot to delete the bucket itself. Resolved!

File: samples/snippets/batch_process_documents_sample_v1beta3_test.py
Comment (leahecole) [inline samples/snippets/batch_process_documents_sample_v1beta3_test.py:38]:
```suggestion
    bucket = storage_client.create_bucket(gcs_output_uri)
```

File: samples/snippets/batch_process_documents_sample_v1beta3_test.py
Comment (leahecole) [inline samples/snippets/batch_process_documents_sample_v1beta3_test.py:32]:
```suggestion
gcs_output_uri_prefix = uuid4()
gcs_output_uri = f"gs://document-ai-python-{gcs_output_uri_prefix}"

```

File: samples/snippets/batch_process_documents_sample_v1beta3_test.py
Comment (leahecole) [inline samples/snippets/batch_process_documents_sample_v1beta3_test.py:62]:
```suggestion
    assert "Paragraph" in out
    assert "Invoice" in out
```

Comment (leahecole) [reply samples/snippets/batch_process_documents_sample_v1beta3_test.py:62]: I looked at the PDF and saw that this word occurred somewhere on it - I wanted to add a check for your file specifically so we know it worked

Comment (leahecole) [reply samples/snippets/batch_process_documents_sample_v1beta3_test.py:62]: This might not be the quite right test, but if you can add a quick check for something specific to the test PDF included, that'd be sweet

File: samples/snippets/process_document_sample_v1beta3_test.py
Comment (leahecole) [inline samples/snippets/process_document_sample_v1beta3_test.py:36]:
```suggestion
    assert "Paragraph" in out
    assert "Invoice" in out
```

Comment (leahecole) [reply samples/snippets/process_document_sample_v1beta3_test.py:36]: This might not be the quite right test, but if you can add a quick check for something specific to the test PDF included, that'd be sweet

File: samples/snippets/quickstart_sample_v1beta3_test.py
Comment (leahecole) [inline samples/snippets/quickstart_sample_v1beta3_test.py:35]:
```suggestion
    assert "Paragraph" in out
    assert "Invoice" in out
```

Comment (leahecole) [reply samples/snippets/quickstart_sample_v1beta3_test.py:35]: This might not be the quite right test, but if you can add a quick check for something specific to the test PDF included, that'd be sweet

**Review by leahecole — CHANGES_REQUESTED** [review]

File: samples/snippets/batch_process_documents_sample_v1beta3_test.py
Comment (aribray) [reply samples/snippets/batch_process_documents_sample_v1beta3_test.py:63]: That worked! Resolved :D

File: samples/snippets/batch_process_documents_sample_v1beta3_test.py
Comment (leahecole) [inline samples/snippets/batch_process_documents_sample_v1beta3_test.py:26]:
```suggestion
```

File: samples/snippets/batch_process_documents_sample_v1beta3_test.py
Comment (leahecole) [inline samples/snippets/batch_process_documents_sample_v1beta3_test.py:32]:
```suggestion
```

File: samples/snippets/batch_process_documents_sample_v1beta3_test.py
Comment (leahecole) [inline samples/snippets/batch_process_documents_sample_v1beta3_test.py:33]:
```suggestion
BUCKET_NAME = f"document-ai-python-{uuid4()}"
```

File: samples/snippets/batch_process_documents_sample_v1beta3_test.py
Comment (leahecole) [inline samples/snippets/batch_process_documents_sample_v1beta3_test.py:38]:
```suggestion
    bucket = storage_client.create_bucket(BUCKET_NAME)
```

File: samples/snippets/batch_process_documents_sample_v1beta3_test.py
Comment (leahecole) [inline samples/snippets/batch_process_documents_sample_v1beta3_test.py:39]:
```suggestion
    yield bucket.name
```

File: samples/snippets/batch_process_documents_sample_v1beta3_test.py
Comment (leahecole) [inline samples/snippets/batch_process_documents_sample_v1beta3_test.py:56]:
```suggestion
        gcs_output_uri=f"gs://{test_bucket}",
```

**Review by leahecole — CHANGES_REQUESTED** [review]

**Review by leahecole — APPROVED** [review]

**Pull request merged**
